subjects/devops/crud-master-py/README.md

```diff
@@ -0,0 +1,318 @@
+send an acknowledgement back.
+
+An example of POST request to
+`http://[API_GATEWAY_URL]:[API_GATEWAY_PORT]/api/billing/`:
+
+```json
+{
+  "user_id": "3",
+  "number_of_items": "5",
+  "total_amount": "180"
+}
+```
+
+> Remember to set up `Content-Type: application/json` for the body of the
+> request.
+
+##### Documenting the API
+
+Good documentation is a very critical feature of every API. By design the APIs
+are meant for others to use, so there have been very good efforts to create
+standard and easy to implement ways to document it.
+
+As an introduction to the art of great documentation you must create an OpenAPI
+documentation file for the API Gateway. There are many different ways to do so,
+a good start could be using SwaggerHub with at least a meaningful description
+for each endpoint. Feel free to implement any extra feature as you see fit.
+
+> You must also create a `README.md` file at the root of your project with
+> detailed instructions on how to build and run your infrastructure and which
+> design choices you made to structure it.
+
+#### Virtual Machines
+
+##### General overview
+
+You will use VirtualBox and Vagrant to set up three different VMs in order to
+test the interactions and correctness of responses between your APIs
+infrastructure.
+
+Vagrant is an open-source software that helps you create and manage virtual
+machines. With Vagrant, you can create a development environment that is
+identical to your production environment, which makes it easier to develop,
+test, and deploy your applications.
+
+Your VMs will be structured as follows:
+
+- `gateway-vm`: This VM will only contain the `api-gateway`.
+- `inventory-vm`: This VM will contain the `inventory-app` API and the database
+  `movies_db`.
+- `billing-vm`: This VM will contain the `billing-app` API, database `orders`
+  and RabbitMQ.
+
+> Vagrant is designed for development and should not be used in production
+> environments.
+
+##### Environment variables
+
+To simplify the building process it is a common practice to store useful
+variables in a `.env` file. It will make very easy to modify/update information
+like URLs, passwords, users and so on.
+
+Those variables will then be used by Vagrant and passed through the different
+microservices in order to centralize all the credentials.
+
+Your `.env` file should contain all the necessary credentials and none of the
+microservices should have any credential hard coded in the source code.
+
+> For the purpose of this exercise the `.env` file must be included in your
+> repository, though usually in public and production projects the `.env` file
+> is never included in repos to avoid sensitive data leaks.
+
+##### Configuration of the VMs
+
+- You will have a `Vagrantfile` which will create and start the three VMs. It
+  will import the environment variables and pass them through each API.
+- You will have a `script/` directory which will store all the scripts you may
```

Comment:
the folder should be named `scripts` cause It contains one or more than 1 script

Reply:
done